src/modm/driver/radio/sx128x_hal_impl.hpp

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) 2023, Rasmus Kleist Hørlyck Sørensen
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_SX128X_HAL_HPP
+#	error  "Don't include this file directly, use 'sx128x_hal.hpp' instead!"
+#endif
+
+namespace modm
+{
+
+template < class SpiMaster, class Nss, class Busy, class Reset >
+Sx128xHalSpi<SpiMaster, Nss, Busy, Reset>::Sx128xHalSpi()
+{
+}
+
+// ----------------------------------------------------------------------------
+
+template < class SpiMaster, class Nss, class Busy, class Reset >
+bool
+Sx128xHalSpi<SpiMaster, Nss, Busy, Reset>::reset()
+{
+    Reset::setOutput();
+    modm::delay_ms(50);
+
+    Reset::setInput();
+    modm::delay_ms(20);
+
+    return true;
```

Comment:
This is going to block the whole system for ~70ms, is that ok?

Reply:
I was debating if I should use a timeout instead. I believe resets would be rare (only upon initialization) so in that aspect it would be ok. However, I will change it to a timeout and use `RF_WAIT_UNTIL(timeout.isExpired())` instead. I have not implmeneted this yet because I will probably put this in Sx128x instead of the transport layer.